packages/palette/src/utils/useLatest.ts

```diff
@@ -0,0 +1,11 @@
+import { useEffect, useRef } from "react"
+
+export const useLatest = <T extends any>(current: T) => {
```

Comment:
This worth adding as a hook, vs just inlining it in the one spot its used?

Reply:
It's actually a pretty clever optimization I've never thought about and want to use it elsewhere